[Felicia Gan] iP

feliciagan opened the pull request:
# Ted 🤖 
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Ted frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is
1. download it from [here](https://github.com/feliciagan/ip/releases/tag/A-Jar).
2. add your tasks.
3. let it manage your tasks for you 😌 

And it is __FREE!__ ✨ 

📝 Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
# 
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

feliciagan committed:
Level-1

feliciagan committed:
Level-2

feliciagan committed:
Level-3

feliciagan committed:
Level-4

feliciagan committed:
A-TextUiTesting

feliciagan committed:
Level-5

feliciagan committed:
Level-6

feliciagan committed:
Implement Level-7 save function

feliciagan committed:
Implement Level-8 date feature

feliciagan committed:
Merge branch 'branch-Level-7'

feliciagan committed:
Resolve merge conflict

feliciagan committed:
Use more OOP

feliciagan committed:
Organise into packages

feliciagan committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

feliciagan committed:
Set up gradle

feliciagan committed:
Add JUnit tests

feliciagan committed:
Change folder and file name to Ted

feliciagan committed:
Change folder and file name to Ted

feliciagan committed:
Package app as JAR file

feliciagan committed:
Add JavaDoc comments to code

feliciagan committed:
Follow the coding standard

feliciagan committed:
Add find task using keyword feature

feliciagan committed:
Merge branch 'branch-A-JavaDoc'

feliciagan committed:
Merge branch-A-CodingStandard

feliciagan committed:
Merge branch-Level-9

feliciagan committed:
Follow coding standard for JUnit test files

GenFusion122 reviewed (commented):
Hey! Your code looks really good, did not find a single coding standard related issue at all! Great Job! 

P.S. Loved how you renamed the bot to Ted!

feliciagan committed:
Add GUI for task bot

Benjamin-Sim commented on src/main/java/ted/task/TaskList.java:
```diff
@@ -0,0 +1,147 @@
+     * @param fw FileWriter that writes to file with saved tasks.
+     * @throws IOException if error is encountered writing to file.
+     */
+    public void writeToFile(FileWriter fw) throws IOException {
+        for (int i = 0; i < tasks.size(); i++) {
+            fw.write(tasks.get(i).toFileString());
+        }
+    }
```
Perhaps moving this method to the Storage class will improve OOP design? This way your file IO logic is completely handled in the Storage class and is abstracted away from other classes.

Benjamin-Sim commented on src/main/java/ted/task/TaskList.java:
```diff
@@ -0,0 +1,147 @@
+    public String list() {
+        String listOfTasks = "";
+        for (int i = 0; i < tasks.size(); i++) {
+            int bulletPoint = i + 1;
```
Not sure I'd call this a bulletPoint since it seems to be the index number of the tasks

Benjamin-Sim commented on src/main/java/ted/task/TaskList.java:
```diff
@@ -0,0 +1,147 @@
+        for (int i = 0; i < tasks.size(); i++) {
+            String toSearch = tasks.get(i).toString();
+            if (toSearch.contains(keyword)) {
+                temp = temp + count + ". " + toSearch + "\n";
```
I notice this logic of formatting the Task objects in a printable list is already present in the list() method above. Perhaps you can modify the list() method to take in the Task objects to print as an argument? That way you can reduce code duplication by calling the list() in this findTasks() method.

Benjamin-Sim reviewed (commented):
Overall good adherence to the naming guidelines. Well done!

I only have some minor suggestions for code organization.

WuHaohui1231 commented on src/main/java/Duke.java:
```diff
@@ -108,6 +183,7 @@ public static void main(String[] args) {
+                    writeToFile();
 
                 } else {
                     throw new DukeException("Oh no, I don't understand T_T\n");
```
Can try to use a `switch` here, as what u did in other parts

WuHaohui1231 reviewed (commented):
The code is excellent overall. It follows the OOP principles

feliciagan committed:
Add assert statements in code

There are no asserts used in the code

Asserts can help programmer spot mistakes in code

Let's add asserts into the code

feliciagan committed:
Improve code quality

Code does not adhere to code quality principles

Improving code quality helps with readability and guards against unsafe practices

Let's improve the code quality

feliciagan committed:
Merge pull request #1 from feliciagan/branch-A-Assertions

Add assert statements in code

feliciagan committed:
Resolve merge conflict

feliciagan committed:
Merge pull request #2 from feliciagan/branch-A-CodeQuality

Improve code quality

feliciagan committed:
Add remind command

No easy way to see upcoming deadlines.

Remind command shows users what deadlines are approaching, helping them prioritise their tasks.

Let's add remind functionality to Ted.

feliciagan committed:
Merge branch 'branch-B-Reminders'

feliciagan committed:
Adhere to coding standard

feliciagan committed:
Update Task list test

feliciagan committed:
Shorten parse method

feliciagan committed:
Acknowledge code reuse

feliciagan committed:
Add more javadoc comments

feliciagan committed:
Add user guide

feliciagan committed:
Update readme

feliciagan committed:
Correct mistake in readme
